packages/manager/src/features/CloudPulse/shared/CloudPulseTagsFilter.tsx

```diff
@@ -0,0 +1,116 @@
+import { Autocomplete } from '@linode/ui';
+import React from 'react';
+
+import { useTagsQuery } from 'src/queries/cloudpulse/tags';
+import { themes } from 'src/utilities/theme';
+
+import type { FilterValue } from '@linode/api-v4';
+
+export interface CloudPulseTags {
+  label: string;
+}
+
+export interface CloudPulseTagsSelectProps {
+  defaultValue?: Partial<FilterValue>;
+  disabled?: boolean;
+  handleTagsChange: (tags: CloudPulseTags[], savePref?: boolean) => void;
+  label: string;
+  placeholder?: string;
+  resourceType: string | undefined;
+  savePreferences?: boolean;
+}
+
+export const CloudPulseTagsSelect = React.memo(
+  (props: CloudPulseTagsSelectProps) => {
+    const {
+      defaultValue,
+      disabled,
+      handleTagsChange,
+      label,
+      placeholder,
+      resourceType,
+      savePreferences,
+    } = props;
+
+    const { data: tags, isError, isLoading } = useTagsQuery(
+      disabled !== undefined ? !disabled : Boolean(resourceType),
+      resourceType,
+      {}
+    );
+
+    const [selectedTags, setSelectedTags] = React.useState<CloudPulseTags[]>();
+
+    const isAutocompleteOpen = React.useRef(false); // Ref to track the open state of Autocomplete
```

Comment:
@kmuddapo do we have to make tags filter work only on the drop down close just like resources?